### ISSUE TITLE
PyRight language server crashes?

### ISSUE THREAD

**Issue by apollo13** [microsoft/pyright#6887]
**Describe the bug**

When changing the "Python > Analysis: Log Level" setting from "Trace" to "Error" I see:
```
[Info  - 11:20:52 AM] Setting pythonPath for service "cloud-init": "/bin/python"
[Info  - 11:20:52 AM] Loading pyproject.toml file at /home/florian/sources/cloud-init/pyproject.toml
[Info  - 11:20:52 AM] Pyproject file "/home/florian/sources/cloud-init/pyproject.toml" has no "[tool.pyright]" section.
[Info  - 11:20:52 AM] Assuming Python version 3.10
Assuming Python platform Linux
DataCloneError: ()=>n could not be cloned.
    at new DOMException (node:internal/per_context/domexception:53:5)
    at Worker.postMessage (node:internal/worker:343:5)
    at c.enqueueRequest (/home/florian/.var/app/com.visualstudio.code/data/vscode/extensions/ms-pyright.pyright-1.1.344/dist/pyright-internal/src/backgroundAnalysisBase.ts:228:26)
    at c.setConfigOptions (/home/florian/.var/app/com.visualstudio.code/data/vscode/extensions/ms-pyright.pyright-1.1.344/dist/pyright-internal/src/backgroundAnalysisBase.ts:58:14)
    at t.BackgroundAnalysisProgram.setConfigOptions (/home/florian/.var/app/com.visualstudio.code/data/vscode/extensions/ms-pyright.pyright-1.1.344/dist/pyright-internal/src/analyzer/backgroundAnalysisProgram.ts:82:35)
    at w.setOptions (/home/florian/.var/app/com.visualstudio.code/data/vscode/extensions/ms-pyright.pyright-1.1.344/dist/pyright-internal/src/analyzer/service.ts:282:41)
    at Function.runWithOptions (/home/florian/.var/app/com.visualstudio.code/data/vscode/extensions/ms-pyright.pyright-1.1.344/dist/pyright-internal/src/languageService/analyzerServiceExecutor.ts:44:27)
    at b.updateOptionsAndRestartService (/home/florian/.var/app/com.visualstudio.code/data/vscode/extensions/ms-pyright.pyright-1.1.344/dist/pyright-internal/src/languageServerBase.ts:525:33)
    at b.updateSettingsForWorkspace (/home/florian/.var/app/com.visualstudio.code/data/vscode/extensions/ms-pyright.pyright-1.1.344/dist/pyright-internal/src/languageServerBase.ts:509:14)
    at async Promise.all (index 0)
[Info  - 11:20:52 AM] Connection to server got closed. Server will restart.
true
[Info  - 11:20:52 AM] Pyright language server 1.1.344 starting
[Info  - 11:20:52 AM] Server root directory: file:///home/florian/.var/app/com.visualstudio.code/data/vscode/extensions/ms-pyright.pyright-1.1.344/dist
[Info  - 11:20:52 AM] Starting service instance "cloud-init"
```
in the logs. Afterwards the language server seems to be hanging.

**VS Code extension or command-line**
Running as VS Code extensions (ms-pyright.pyright-1.1.344) on VSCode 1.84.2

**Comment by erictraut** [comment]
@debonte, @heejaechang, this looks like something related to message-passing to the background worker. Could one of you take a look?

**Comment by rchiodo** [comment]
Yeah it's likely a URI that's being passed across that needs to be turned into plain json.

**Comment by rchiodo** [comment]
@apollo13, is this the repository you're using? https://github.com/canonical/cloud-init

I can't reproduce the problem with just a sample folder.

**Comment by rchiodo** [comment]
It might also be something in your settings.json considering the sending of the config settings is causing the crash. Can you share that too?

**Comment by apollo13** [comment]
@rchiodo Yes that is the repo I am using. My config:
```json
{
    "workbench.startupEditor": "none",
    "python.languageServer": "None",
    "python.analysis.logLevel": "Trace",
}
```
(I am usually using a new configuration to report bugs, python and pyright are the only installed extensions as well)

**Comment by rchiodo** [comment]
Thanks, I was able to reproduce the problem. I forgot that debugging Pyright turns off the background thread.

**Comment by rchiodo** [comment]
Fix should be in this weeks release, I was able to reproduce and fix the problem. Sorry for the crash :)

**Comment by apollo13** [comment]
No biggie, it was annoying enough that I disabled one extension after the other to find it so I could work again. Thank you for fixing it 👍

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.345, which I just published. It will also be included in a future release of pylance.
